sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -295,10 +304,10 @@ private SpecializedBlobClientBuilder prepareBuilder() {
             final ParallelTransferOptions finalParallelTransferOptions = parallelTransferOptions == null
                 ? new ParallelTransferOptions(null, null, null)
                 : new ParallelTransferOptions(parallelTransferOptions.getBlockSize(),
-                    parallelTransferOptions.getNumBuffers(), parallelTransferOptions.getProgressReceiver());
+                parallelTransferOptions.getNumBuffers(), parallelTransferOptions.getProgressReceiver());
 
             // See ProgressReporter for an explanation on why this lock is necessary and why we use AtomicLong.
-            AtomicLong totalProgress = new AtomicLong(0);
+            AtomicLong totalProgress = new AtomicLong();
```

Comment:
L354 and L364 (path 1) call  `getBlockBlobAsyncClient()` twice which is a bit of an expensive operation. Do the `get` once and use it in path 1 and path 2.